tests/share/test_harvest.py

```diff
@@ -401,3 +401,45 @@ def test_harvest_after(self, monkeypatch, now, end_date, harvest_after, should_r
         tasks.harvest()
 
         assert source_config.harvester.get_class()._do_fetch.called == should_run
+
+    def test_latest_date(self):
+        source_config = factories.SourceConfigFactory(
+            full_harvest=True,
+            earliest_date=pendulum.parse('2017-01-01').date()
+        )
+
+        # We have a harvest log with start_date equal to earliest_date
+        # but a different source_config
+        factories.HarvestLogFactory(
+            start_date=pendulum.parse('2017-01-01').date(),
+            end_date=pendulum.parse('2017-01-02').date(),
+        )
+
+        assert len(HarvestScheduler(source_config).all(cutoff=pendulum.parse('2018-01-01').date())) == 365
+
+    def test_caught_up(self):
+        source_config = factories.SourceConfigFactory(
+            full_harvest=True,
+            earliest_date=pendulum.parse('2017-01-01').date()
+        )
+
+        factories.HarvestLogFactory(
+            source_config=source_config,
+            start_date=pendulum.parse('2017-01-01').date(),
+            end_date=pendulum.parse('2017-01-02').date(),
+        )
+
+        factories.HarvestLogFactory(
+            source_config=source_config,
+            start_date=pendulum.parse('2018-01-01').date(),
+            end_date=pendulum.parse('2018-01-02').date(),
+        )
+
+        assert len(HarvestScheduler(source_config).all(cutoff=pendulum.parse('2018-01-01').date())) == 0
```

Comment:
I realize this isn't new, but it seems odd (or at least unintuitive) that `all()` doesn't fill in any gaps, it just goes from the most recent HarvestLog until the cutoff. I don't know that I would rename it (because I can't think of a better name), but I think it'd be worth mentioning that in the docstring.